### ISSUE TITLE
How is the Agones installation tested on different cloud platforms?

### ISSUE THREAD

**Issue by Yingxin-Jiang** [googleforgames/agones#1105]
This is more of a question rather than feature request. According to https://agones.dev/site/docs/installation/, Agones can be installed on GKE, Minikube, EKS, or AKS. How is the Agones installation tested on different cloud platforms? Is it a manual process? If so, does it make sense to include it in e2e tests?

Thanks!

**Comment by roberthbailey** [comment]
Right now the automated testing is all done on Google Cloud (using GKE). Testing on other platforms is done manually by contributors and users. Note that when we upgrade to a new version of Kubernetes (e.g. the recent migration from [1.11 to 1.12](https://github.com/googleforgames/agones/issues/717)) we did some manual testing (not as rigorous as the e2e suite though) on all of the supported platforms.

**Comment by roberthbailey** [comment]
It would be awesome to add more automated testing on other platforms. 

One issue is that to test on other cloud we need an account with billing configured. We get this "for free" for GCP right now.

Another issue is that the current presubmit and e2e tests are run through google cloud build. I've started working on [getting prow configured](https://github.com/googleforgames/agones/issues/649) which would much more easily allow us to write tests against other cloud providers. 

For both of these issues (and others I surely haven't thought of), help would be greatly appreciated!

**Comment by Yingxin-Jiang** [comment]
Thanks @roberthbailey for the info! I will leave the issue open for adding more automated testing.

In addition, it would be helpful if we can document how to install Agones using Terraform for EKS. I didn't see it in https://agones.dev/site/docs/installation/terraform/. Thanks!

**Comment by roberthbailey** [comment]
The [PR to add it to the docs](https://github.com/googleforgames/agones/pull/986) is currently marked as draft. Once @aLekSer is ready we will get it reviewed and merged but you can check out the draft in the mean time.

**Comment by roberthbailey** [comment]
Right now we aren't prioritizing adding automated testing on other providers, so I'm going to re-label this as a question and mark it as closed.
